### PR TITLE
Fixed unselcting error on clicking infobar

### PR DESCRIPTION
**Pull request by kasvith**
Fixed the unselect on clicking infobar

Pull Request for Issue #515  .

### Summary of Changes
- Added more control to the media browser

### Testing Instructions
- Click on a file
- Open media infobar
- Click on media infobar

### Expected result
- Media Infobar should be still open

### Actual result
- It is actual open until you click elsewhere in browser to deselect

### CODE REVIEW

**Comment by dneukirchen** [comment]
thank you @kasvith 
please do not use $root, $parent, $children and implement the solution we talked about in glip.

**Comment by kasvith** [comment]
Sure, so we can work on a solution for overcome this. thanks @dneukirchen

**Comment by kasvith** [comment]
is this fixed? when i pulled from main repo it seems fixed somehow, or maybe my browser fools me. Tried with clearing caches also

**Comment by kasvith** [comment]
Issue seems resolved somehow, im closing this for now

**Comment by kasvith** [comment]
Sorry it was my browser cache :D

**Comment by kasvith** [comment]
@dneukirchen following changes are done

As we use **infobar** as a child in reality (in visual context) I ve moved that to **MediaBrowser** as a child component

Now when we want to fix this **unselect** error its being used with **refs** as we do for other media elements in media browser.

I prefer this way as we dont have to pass props to media browser

File: administrator/components/com_media/resources/scripts/components/browser/browser.vue [hunk 1]
```diff
@@ -64,12 +67,16 @@
...
+            isModal() {
+		        return Joomla.getOptions('com_media', {}).isModal;
             }
         },
         methods: {
             /* Unselect all browser items */
             unselectAllBrowserItems(event) {
-                const eventOutside = (this.$refs.browserItems && !this.$refs.browserItems.contains(event.target)) || event.target === this.$refs.browserItems;
+                const eventOutside = (this.$refs.browserItems && !this.$refs.browserItems.contains(event.target)) &&
+                                    event.path.indexOf(this.$refs.infobar.$el) < 0 || event.target === this.$refs.browserItems;
```
Comment (dneukirchen) [inline administrator/components/com_media/resources/scripts/components/browser/browser.vue]: Please check the conditions.

Comment (kasvith) [reply administrator/components/com_media/resources/scripts/components/browser/browser.vue]: i split that into two lines for better readiability

Comment (kasvith) [reply administrator/components/com_media/resources/scripts/components/browser/browser.vue]: fixed

**Comment by kasvith** [comment]
PR updated

**Review by dneukirchen — CHANGES_REQUESTED** [review]
Thats not what i meant:

the second `&&` should be a `||` condition and `$refs.infobar.$el` might be undefined. 
Open infobar, close infobar, click anywhere, see errors in console.

**Comment by laoneo** [comment]
I guess you need to compile the files with:
npm run build

**Comment by kasvith** [comment]
Will look into that

**Comment by kasvith** [comment]
Done updated with build

@dneukirchen I dont see any error in my console window. **infobar** is defined as a ref :)

**Comment by kasvith** [comment]
turning second && to || made browser not selecting any items until few clicks

**Review by dneukirchen — APPROVED** [review]

**Comment by dneukirchen** [comment]
thx @kasvith gj

**Comment by kasvith** [comment]
Finally :D

**Pull request merged**